### PR TITLE
Upgrade GitHub Workflow to use latest Ubuntu

### PR DESCRIPTION
**Pull request by chetangudisagar**
## Overview
Ubuntu image version upgrade in GitHub Actions.

Description:
Update image label to ubuntu-latest.
Fix https://github.com/actions/runner-images/issues/6002 to ensure continued support.

Why should this be merged: Fixes the issue.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Review by souravgarg — COMMENTED** [review]
QQ. 18.04 ubuntu version is LTS. Do we have such requirements on LTS releases to be used in tests?

**Comment by chetangudisagar** [comment]
> QQ. 18.04 ubuntu version is LTS. Do we have such requirements on LTS releases to be used in tests?

I think they would point `ubuntu-latest` to an LTS version(20.04 as seen below). And I don't think we have any specific requirements for these versions from a product perspective.

> Injection Framework Test: .github#L1 The ubuntu-18.04 environment is deprecated, consider switching to ubuntu-20.04(ubuntu-latest), or ubuntu-22.04 instead. For more details see https://github.com/actions/virtual-environments/issues/6002

File: .github/workflows/compatibility-test.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on: pull_request
 jobs:
   compatibility_test:
     # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (Maithem) [inline .github/workflows/compatibility-test.yml:8]: Does the latest ubuntu version contain jdk 1.8?

Comment (chetangudisagar) [reply .github/workflows/compatibility-test.yml:8]: Good Question!

Just checked that it does have it.
https://github.com/actions/runner-images#available-images `ubuntu-latest` points to 22.04,

and 22.04 runner image has Java 8 -
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#java

**Review by Maithem — APPROVED** [review]

**Pull request merged**